crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -38,6 +41,263 @@ async fn local_node_price_improvement_fee_sell_order() {
+        buy_amount: to_wei(6),
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: OrderKind::Sell,
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader.private_key()).unwrap()),
+    );
+    let partner_fee_order = OrderCreation {
+        sell_token: onchain.contracts().weth.address(),
+        sell_amount: to_wei(5),
+        buy_token: partner_fee_order_token.address(),
+        buy_amount: to_wei(3),
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: OrderKind::Sell,
+        app_data: partner_fee_app_data,
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader.private_key()).unwrap()),
+    );
+
+    let market_price_improvement_uid = services
+        .create_order(&market_price_improvement_order)
+        .await
+        .unwrap();
+    let limit_surplus_order_uid = services.create_order(&limit_surplus_order).await.unwrap();
+    let partner_fee_order_uid = services.create_order(&partner_fee_order).await.unwrap();
+
+    let mut config = vec![
+        "--drivers=test_solver|http://localhost:11088/test_solver".to_string(),
+        "--price-estimation-drivers=test_quoter|http://localhost:11088/test_solver".to_string(),
+    ];
+    config.extend(autopilot_config);
+    services.start_autopilot(None, config);
+
+    tracing::info!("Acquiring a quote.");
+    let quote_request = OrderQuoteRequest {
+        sell_token: onchain.contracts().weth.address(),
+        buy_token: limit_order_token.address(),
+        side: OrderQuoteSide::Sell {
+            sell_amount: SellAmount::AfterFee {
+                value: NonZeroU256::try_from(5).unwrap(),
+            },
+        },
+        validity: Validity::To(model::time::now_in_epoch_seconds() + 300),
+        ..Default::default()
+    };
+    let quote = services.submit_quote(&quote_request).await.unwrap();
+
+    tracing::info!("Rebalancing AMM pool.");
+    onchain
+        .mint_token_to_weth_uni_v2_pool(&limit_order_token, to_wei(1000))
+        .await;
+
+    tracing::info!("Waiting for liquidity state to update");
+    wait_for_condition(TIMEOUT, || async {
+        // Mint blocks until we evict the cached liquidity and fetch the new state.
+        onchain.mint_block().await;
+        let next = services.submit_quote(&quote_request).await.unwrap();
+        next.quote.buy_amount != quote.quote.buy_amount
+    })
+    .await
+    .unwrap();
+
+    tracing::info!("Waiting for trade.");
+    wait_for_condition(TIMEOUT, || async { services.solvable_orders().await == 3 })
+        .await
+        .unwrap();
+
+    wait_for_condition(TIMEOUT, || async { services.solvable_orders().await == 0 })
+        .await
+        .unwrap();
+
```

Comment:
Why are all of these waits important? Couldn't we just wait for the limit order to trade? This could make this section shorter to read.

Reply:
Since starting the driver's solving endpoint isn't straightforward, replaced with the following assertion which should probably wait until all the orders are executed and filled: https://github.com/cowprotocol/services/pull/2561/commits/54507fcb7877bca594329b0f7f2d63f3558f5d7b